articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,455 @@
+Note that the terminal's directory location needs to be the same as the Q# file, so we recommend simply using the integrated terminal in VS Code or Visual Studio.
+
+### Add entry point to Q# file
+Of course, most Q# files will contain more than one operation.
+So, we need to let the compiler know *which* operation to execute when we provide the `dotnet run` command.
+
+This is done with a simple change to the Q# file itself: add a line with `@EntryPoint()` directly preceding the operation.
+
+Our file from above would therefore become
+```qsharp
+namespace NamespaceName {
+    open Microsoft.Quantum.Intrinsic;     // for the H operation
+    open Microsoft.Quantum.Measurement;   // for MResetZ
+
+	@EntryPoint()
+    operation MeasureSuperposition() : Result {
+        using (q = Qubit()) { 
+            H(q);
+            return MResetZ(q);
+        }
+    }
+}
+```
+
+Now, a call of `dotnet run` from the command line leads to `MeasureSuperposition` being run, and the returned value is then printed directly to the terminal.
+So, you will see either `One` or `Zero` printed. 
+
+Note that it doesn't matter if you have more operations defined below it, only `MeasureSuperposition` will be run.
+Additionally, it's no problem if your operation includes [documentation comments](xref:microsoft.quantum.guide.filestructure#documentation-comments) before the operation declaration, the `@EntryPoint()` attribute can be simply placed above them.
+
+### Operation arguments
+
+So far, we've only considered an operation that takes no arguments.
+Suppose we wanted to perform a similar operation, but on multiple qubits---the number of which is provided as an argument.
+Such an operation could be written as
+```qsharp
+    operation MeasureSuperpositionArray(n : Int) : Result[] {
+        mutable resultArray = new Result[n];     // instantiate a length n array for the measurement Results 
+        using (qubits = Qubit[n]) {              // allocate a register of n qubits
+            ApplyToEach( H , qubits);            // apply H to each qubit in the register (from Microsoft.Quantum.Canon)
+            set resultArray = MultiM(qubits);    // measure the array of qubits
+            ResetAll(qubits);                    // reset all the qubits
+        }
+        return resultArray;
+    }
+```
+where the returned value is an array of the measurement results.
+Note that the [`ApplyToEach`](xref:microsoft.quantum.canon.applytoeach) operation is in the `Microsoft.Quantum.Canon` namespace, requiring another `open` statement with the others.
+
+If we move the `@EntryPoint()` to precede this new operation (note there can only be one such line in a file), attempting to run it with simply `dotnet run` results in a helpful error message.
+This message indicates what additional command line options are required, and how to express them.
+
+The general format for the command line is actually `dotnet run [options]`, and operation arguments are provided there.
```

Comment:
Should we mention `--` here? If you have a Q# input called `noBuild`, for instance, that will get turned into `--no-build`, which is a valid option for `dotnet run` such that you need `--` to separate options for `dotnet run` from the options for the program being run.

Reply:
I encountered that when I was exploring the `dotnet run` options with trial-and-error; specifically when I tried using `numQubits` as the Q# input instead of `n`. Completely forgot to inquire more, but I think it's definitely something we should mention. Is there more comprehensive documentation for it anywhere that I can get info from? E.g. does it work from camel case and therefore would turn `numQubitsOne` into `--num-qubits-one`?

Reply:
I think @samarsha may be able to provide more info than I can here. As for where `--` is documented, that's a `dotnet run` thing rather than a Q# thing; if you run Q# standalone executable outside of the .NET Core CLI, then `--` doesn't come up. Thus, I think linking to the [.NET Core CLI docs on `--`](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-run#options) could be helpful to explain why `--` is needed.

Reply:
Yes, the standalone executable will convert `camelCase` parameter names to `--kebab-case` option names.